rai/test_utils.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 RelationalAI, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rai
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func NewTestClient() (*Client, error) {
+	configPath, _ := expandUser(DefaultConfigFile)
+	if _, err := os.Stat(configPath); err == nil {
+		return NewDefaultClient()
+	}
+
+	var cfg Config
+
+	clientId := os.Getenv("CLIENT_ID")
+	clientSecret := os.Getenv("CLIENT_SECRET")
+	clientCredentialsUrl := os.Getenv("CLIENT_CREDENTIALS_URL")
+
+	placeHolderConfig := `
+		[default]
+		host=azure.relationalai.com
+		region=us-east
+		port=443
+		scheme=https
+		client_id=%s
+		client_secret=%s
+		client_credentials_url=%s
+	`
+	configSrc := fmt.Sprintf(placeHolderConfig, clientId, clientSecret, clientCredentialsUrl)
+	LoadConfigString(configSrc, "default", &cfg)
+	opts := ClientOptions{Config: cfg}
+	return NewClient(context.Background(), &opts), nil
+}
+
+// Answers if the given list contains the given value
+func contains(items []string, value string) bool {
+	for _, v := range items {
+		if v == value {
+			return true
+		}
+	}
+	return false
+}
+
+func isErrNotFound(err error) bool {
+	e, ok := err.(HTTPError)
+	if !ok {
+		return false
+	}
+	return e.StatusCode == http.StatusNotFound
+}
+
+// Ensure that the test database exists.
+func EnsureDatabase(t *testing.T, client *Client, dbname string) {
+	if _, err := client.GetDatabase(dbname); err != nil {
+		assert.True(t, isErrNotFound(err))
+		_, err := client.CreateDatabase(dbname)
+		assert.Nil(t, err)
+	}
+}
+
+// Ensure that the test engine exists.
+func EnsureEngine(t *testing.T, client *Client, engine string) {
+	if _, err := client.GetEngine(engine); err != nil {
+		assert.True(t, isErrNotFound(err))
+		_, err = client.CreateEngine(engine, "XS")
+		assert.Nil(t, err)
+	}
+}
+
+func TearDownEngine(client *Client, engine string) {
```

Comment:
you dont need the wrapper if its just a single line .. this just makes the code harder to follow